python/ray/autoscaler/autoscaler.py

```diff
@@ -67,9 +67,11 @@ def __init__(self,
             self.instance_types = self.config["available_instance_types"]
             self.resource_demand_scheduler = ResourceDemandScheduler(
                 self.provider, self.instance_types, self.config["max_workers"])
+            self.node_resource_mapping = {}
```

Comment:
Isn't this redundant with self.instance_types?

Reply:
this is a node_id -> resources mapping. instance type is ambiguous if we have multiple nodes with the same instance type but different custom resources